cassandra/io/asyncorereactor.py

```diff
@@ -86,24 +88,22 @@ def maybe_start(self):
     def _run_loop(self):
         log.debug("Starting asyncore event loop")
         with self._loop_lock:
-            while True:
+            while not self._shutdown:
                 try:
-                    asyncore.loop(timeout=0.001, use_poll=True, count=1000)
+                    asyncore.loop(timeout=0.001, use_poll=True, count=100)
+                    self._timers.service_timeouts()
+                    if not asyncore.socket_map:
+                        time.sleep(0.005)
```

Comment:
It's probably not worth a lot of effort to use a non-global map.  The only thing I would maybe check is whether that attr is present and used in Jython.  (I know that Jython isn't really supported right now, but 2.7 was just released, so it would be good to know if there's obvious breakage.)